packages/core/src/App/Containers/AccountSignupModal/account-signup-modal.jsx

```diff
@@ -113,6 +119,56 @@ const AccountSignup = ({
         setABQuestionnaire(fetchQuestionnarieData());
     }, []); // eslint-disable-line react-hooks/exhaustive-deps
 
+    React.useEffect(() => {
+        if (is_signup_flow_error) {
+            cacheTrackEvents.trackConsoleErrors(errorMessage => {
+                if (errorMessage) {
+                    const screen_name = !is_password_modal ? 'country_selection_screen' : 'password_screen_opened';
+
+                    // Check and set the logging state using the ref
+                    if (screen_name === 'country_selection_screen' && !isCountryScreenLoggedOnceRef.current) {
+                        cacheTrackEvents.loadEvent([
+                            {
+                                event: {
+                                    name: 'ce_virtual_signup_form',
+                                    properties: {
+                                        action: 'signup_flow_error',
+                                        form_name: is_mobile
+                                            ? 'virtual_signup_web_mobile_default'
+                                            : 'virtual_signup_web_desktop_default',
+                                        error_message: localize(errorMessage),
+                                        screen_name,
+                                    },
+                                },
+                                cache: true,
+                            },
+                        ]);
+
+                        // Update both the ref and state
+                        isCountryScreenLoggedOnceRef.current = true;
+                    } else if (screen_name === 'password_screen_opened') {
+                        cacheTrackEvents.loadEvent([
+                            {
+                                event: {
+                                    name: 'ce_virtual_signup_form',
+                                    properties: {
+                                        action: 'signup_flow_error',
+                                        form_name: is_mobile
+                                            ? 'virtual_signup_web_mobile_default'
+                                            : 'virtual_signup_web_desktop_default',
+                                        error_message: localize(errorMessage),
+                                        screen_name,
+                                    },
+                                },
+                                cache: true,
+                            },
+                        ]);
+                    }
```

Comment:
The only difference of the 2 condition is `screen_name`. can we add the condition there instead of creating different condition blocks?

Reply:
The is_password_modal state currently manages both the country selection and password screens. However, when an error occurs on either screen, it causes the screen name for the country screen to be duplicated in the logs. For example:
```
[
    {"name": "ce_virtual_signup_form", "properties": {"action": "signup_flow_error", "form_name": "virtual_signup_web_desktop_default", "error_message": "country", "screen_name": "country_selection_screen"}},
    {"name": "ce_virtual_signup_form", "properties": {"action": "signup_flow_error", "form_name": "virtual_signup_web_desktop_default", "error_message": "password", "screen_name": "country_selection_screen"}},
    {"name": "ce_virtual_signup_form", "properties": {"action": "signup_flow_error", "form_name": "virtual_signup_web_desktop_default", "error_message": "password", "screen_name": "password_screen_opened"}}
]
```
To resolve the issue of logging country_selection_screen multiple times, we need to ensure that this screen name is only recorded the first time an error occurs on that screen. After the initial error, only subsequent errors related to the password_screen_opened screen should be logged, until the password modal is triggered.
That is the reason we added the two conditions.Let me know if you have any other solution for it

Reply:
Then we can put this in a function so we don't need to duplicate 

```
cacheTrackEvents.loadEvent([
      {
          event: {
              name: 'ce_virtual_signup_form',
              properties: {
                  action: 'signup_flow_error',
                  form_name: is_mobile
                      ? 'virtual_signup_web_mobile_default'
                      : 'virtual_signup_web_desktop_default',
                  error_message: localize(errorMessage),
                  screen_name,
              },
          },
          cache: true,
      },
  ]);
```

Reply:
okay i will add this inside a function